dd-java-agent/agent-iast/src/main/java/com/datadog/iast/securitycontrol/IastSecurityControlTransformer.java

```diff
@@ -0,0 +1,61 @@
+package com.datadog.iast.securitycontrol;
+
+import datadog.trace.api.iast.securitycontrol.SecurityControl;
+import java.lang.instrument.ClassFileTransformer;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IastSecurityControlTransformer implements ClassFileTransformer {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(IastSecurityControlTransformer.class);
+
+  private final List<SecurityControl> securityControls;
+  private final Set<String> classFilter;
+
+  public IastSecurityControlTransformer(List<SecurityControl> securityControls) {
+    this.securityControls = securityControls;
+    this.classFilter =
+        securityControls.stream().map(SecurityControl::getClassName).collect(Collectors.toSet());
+  }
+
+  @Override
+  @Nullable
+  public byte[] transform(
+      ClassLoader loader,
+      String className,
+      Class<?> classBeingRedefined,
+      java.security.ProtectionDomain protectionDomain,
+      byte[] classfileBuffer) {
+    if (!classFilter.contains(className)) {
+      return null; // Do not transform classes that are not in the classFilter
+    }
+    List<SecurityControl> match = getSecurityControl(className);
+    if (match == null || match.isEmpty()) {
+      return null; // Do not transform classes that do not have a security control
+    }
+    try {
+      ClassReader cr = new ClassReader(classfileBuffer);
+      ClassWriter cw = new ClassWriter(cr, ClassWriter.COMPUTE_FRAMES);
+      ClassVisitor cv = new SecurityControlMethodClassVisitor(cw, match);
+      cr.accept(cv, 0);
```

Comment:
```suggestion
      cr.accept(cv, SKIP_DEBUG | SKIP_FRAMES);
```

I think you just need to visit the code, but you can safely ignore debug info and frames.